### ISSUE TITLE
LDAP Authentication ClassCastException

### ISSUE THREAD

**Issue by cpcundill** [pac4j/play-pac4j#83]
Hey there,

Firstly, thanks for the library - nice to see something Java-based for Play which can do LDAP.

Unfortunately I'm running into an error when trying to use the LdapAuthenticator with the FormClient:

```
ClassCastException: org.pac4j.ldap.profile.LdapProfile cannot be cast to org.pac4j.http.profile.HttpProfile]
```

Looking through the code, I can see this is because the LdapAuthenticator creates an LdapProfile yet the CallbackController tries to get the profile via the client, which is the `FormClient`.  This in turn, uses it's superclass `IndirectHttpClient` to `retrieveUserProfile`.  The `IndirectHttpClient` is tied to the `HttpProfile` and thus expects to receive this type but doesn't and then blows up.

Is this just my mis-configuration or a genuine problem?  Should I be using a different client to `FormClient` when using the `LdapAuthenticator`?

**Comment by leleuj** [comment]
You caught a bug: indeed, the `FormClient` works with a `HttpProfile` and the `LdapProfile` should inherit from a `HttpProfile`.
I just fixed everything right now on the 1.8.0-RC2-SNAPSHOT: would you mind doing a test?
The final release is planned for this Thursday.

**Comment by cpcundill** [comment]
Thanks for addressing this so quickly, will try with the RC2-SNAPSHOT.

**Comment by cpcundill** [comment]
BTW, since the fix introduces "artificial" inheritance, i.e. `LdapProfile` extends `HttpProfile`, you may want to place some refactoring on your roadmap.

**Comment by leleuj** [comment]
I would not say that the inheritance is "artificial": `UserProfile` > `CommonProfile` > `HttpProfile` > `LdapProfile`. It's an easy way to sort profiles.
If you feel that there is some room for improvement, don't hesitate to open an issue: I'm open to suggestion (for version 1.9)...

**Comment by cpcundill** [comment]
Might just do that.  Thanks.

**Comment by cpcundill** [comment]
Hey there @leleuj just to say that the fix in 1.8.0-RC2-SNAPSHOT worked for me.

**Comment by leleuj** [comment]
Great!
